tools/distpartitioning/globalids.py

```diff
@@ -73,9 +50,8 @@ def get_shuffle_global_nids(rank, world_size, global_nids_ranks, node_data):
             send_nodes.append(torch.empty((0), dtype=torch.int64))
 
     #send receive global-ids
-    alltoallv_cpu(rank, world_size, recv_shuffle_global_nids, send_nodes)
-
-    shuffle_global_nids = np.concatenate([x.numpy() for x in recv_shuffle_global_nids])
+    recv_shuffle_global_nids = alltoallv_cpu_data(rank, world_size, send_nodes, torch.int64)
+    shuffle_global_nids = np.concatenate([x.numpy() if x != None else [] for x in recv_shuffle_global_nids])
```

Comment:
should we use `if x is not None`?

Reply:
did you test the case? if one of the element is `[]` in the input list, does numpy.concatenate works?

Reply:
Code changed to use 'if x is not None:'

Yes this case is specifically tested. This happens in the case of "mag" dataset with cyclic partitioning. The numpy.concatenate works when one of the list elements is '[]'.

Reply:
i don't see you make such a change.

Reply:
Earlier I misunderstood this comment. Now changed the code appropriately to match the description in the CI comment.